test/performance/performance.go

```diff
@@ -0,0 +1,333 @@
+	pflag.IntVarP(&n_total, "total-policies", "t", 1000, "total number of policies created")
+	pflag.IntVarP(&per_batch_sleep, "sleep", "s", 20, "time (min) to sleep after creating a batch of policies")
+	pflag.StringVar(&output_filename, "csv", performance_dir+"results.csv", "path to CSV to export results to")
+
+	pflag.Parse()
+
+	// wait for config map to be applied properly before executing command
+	_, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token")
+	if err != nil {
+		klog.V(2).Info("Setting up secret for k8s metrics")
+		_, err = exec.Command("kubectl", "apply", "-f", performance_dir+"/resources/setup/metrics-configmap.yaml").CombinedOutput()
+		if err != nil {
+			klog.Exitf("Error applying metrics configMap: %s", err)
+		}
+		// wait for secret to be created for 2 minutes
+		found := false
+		for i := 0; i < 120; i += 5 {
+			_, err = bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token")
+			if err == nil {
+				found = true
+				break
+			}
+			klog.V(2).Info("Secret not found, waiting 5 seconds...")
+			time.Sleep(5 * time.Second)
+		}
+
+		if !found {
+			klog.Fatal("Error: Prometheus secret creation timed out")
+		}
+	}
+
+	secret, err := bash("oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'")
+	if err != nil {
+		klog.Exitf("Error getting Prometheus metrics secret: %s", err)
+	}
+
+	token, err := bash(fmt.Sprintf("echo $(oc get secret %s -n openshift-user-workload-monitoring -o json | jq -r '.data.token') | base64 -d", secret))
+	if err != nil {
+		klog.Exitf("Error getting API token from secret: %s", err)
+	}
+
+	thanos_host, err := bash("oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'")
+	if err != nil {
+		klog.Exitf("Error getting Thanos host: %s", err)
+	}
+
+	klog.Info("Setup complete! Starting the Config Policy Controller performance test :)")
+	table_data := []metricData{}
+
+	table_data = append(table_data, getMetrics(thanos_host, token, per_batch_sleep, 0))
+
+	for total_plcs := 0; total_plcs < n_total; total_plcs += n_per_batch {
+		start := time.Now()
+
+		// create a batch of policies
+		klog.V(2).Infof("Creating %d copies of %s on the managed cluster...\n", n_per_batch, plc_filename)
+		batch_fails := 0
+		for batch_plcs := 0; batch_plcs < n_per_batch; batch_plcs += 1 {
+			sed, err := bash(fmt.Sprintf(`sed "s/\[ID\]/%s/" %s > %s/resources/current_policy.yaml`, uuid.New().String(), performance_dir+"/"+plc_filename, performance_dir))
+			if err != nil {
+				klog.Exitf("Error patching policy with unique name: %s; %s", err, sed)
+			}
+
+			creation_output, err := exec.Command("oc", "apply", "-f", performance_dir+"/resources/current_policy.yaml").CombinedOutput()
+			if err != nil {
+				klog.Error("Error creating policy: %s, %s", err, string(creation_output))
+				batch_fails += 1
```

Comment:
My initial thought is to add a retry on failure and if it still fails, we can exit the script. I just worry about having inconsistent data if the number of policies is different per run.